File: AspNetMvcUnleashed/0672329980_source/Chapter20/VB/UnleashedBlog/UnleashedBlog/Helpers/BlogPagerHelper.vb

Namespace Helpers

    Public Module BlogPagerHelper

        <System.Runtime.CompilerServices.Extension()> _
        Public Function BlogPager(ByVal helper As HtmlHelper, ByVal pager As IPagedList) As String
            ' Don't display anything if not multiple pages
            If pager.TotalPageCount = 1 Then
                Return String.Empty
            End If

            ' Build route data
            Dim routeData = New RouteValueDictionary(helper.ViewContext.RouteData.Values)

            ' Build string
            Dim sb = New StringBuilder()

            ' Render Newer Entries
            If pager.PageIndex > 0 Then
                routeData("page") = pager.PageIndex - 1
                sb.Append(helper.ActionLink("Newer Entries", "Index", routeData))
            End If

            ' Render divider
            If pager.PageIndex > 0 AndAlso pager.PageIndex < pager.TotalPageCount - 1 Then
                sb.Append(" | ")
            End If

            ' Render Older Entries
            If pager.PageIndex < pager.TotalPageCount - 1 Then
                routeData("page") = pager.PageIndex + 1
                sb.Append(helper.ActionLink("Older Entries", "Index", routeData))
            End If

            Return sb.ToString()
        End Function

        <System.Runtime.CompilerServices.Extension()> _
        Public Function BlogPager(ByVal helper As AjaxHelper, ByVal pager As IPagedList) As String
            ' Don't display anything if not multiple pages or no more entries
            If pager.TotalPageCount = 1 OrElse pager.PageIndex = pager.TotalPageCount - 1 Then
                Return String.Empty
            End If

            ' Build route data
            Dim routeData = New RouteValueDictionary(helper.ViewContext.RouteData.Values)

            ' Build Ajax options
            Dim options = New AjaxOptions With {.UpdateTargetId = "blogEntries", .InsertionMode = InsertionMode.InsertAfter, .LoadingElementId = "loadingMoreEntries", .OnBegin = "function() {this.style.display='none';}"}

            ' Render More Entries
            routeData("page") = pager.PageIndex + 1
            Return helper.ActionLink("More Entries", "Index", routeData, options)
        End Function

    End Module
End Namespace

File: AspNetMvcUnleashed/0672329980_source/Chapter21/CS/UnleashedBlog/UnleashedBlog/Helpers/BlogPagerHelper.cs
using System.Text;
using System.Web.Mvc;
using System.Web.Mvc.Html;
using System.Web.Mvc.Ajax;
using System.Web.Routing;
using Paging;
using System.Collections.Generic;

namespace UnleashedBlog.Helpers
{
    public static class BlogPagerHelper
    {

        public static string BlogPager(this HtmlHelper helper, IPagedList pager)
        {
            // Don't display anything if not multiple pages
            if (pager.TotalPageCount == 1)
                return string.Empty;

            // Build route data
            var routeData = new RouteValueDictionary(helper.ViewContext.RouteData.Values);

            // Build string
            var sb = new StringBuilder();

            // Render Newer Entries
            if (pager.PageIndex > 0)
            {
                routeData["page"] = pager.PageIndex - 1;
                sb.Append(helper.ActionLink("Newer Entries", "Index", routeData));
            }

            // Render divider
            if (pager.PageIndex > 0 && pager.PageIndex < pager.TotalPageCount - 1)
                sb.Append(" | ");

            // Render Older Entries
            if (pager.PageIndex < pager.TotalPageCount - 1)
            {
                routeData["page"] = pager.PageIndex + 1;
                sb.Append(helper.ActionLink("Older Entries", "Index", routeData));
            }

            return sb.ToString();
        }


        public static string BlogPager(this AjaxHelper helper, IPagedList pager)
        {
            // Don't display anything if not multiple pages or no more entries
            if (pager.TotalPageCount == 1 || pager.PageIndex == pager.TotalPageCount -1)
                return string.Empty;

            // Build route data
            var routeData = new RouteValueDictionary(helper.ViewContext.RouteData.Values);

            // Build Ajax options
            var options = new AjaxOptions 
                { 
                    UpdateTargetId = "blogEntries", 
                    InsertionMode=InsertionMode.InsertAfter, 
                    LoadingElementId="loadingMoreEntries",
                    OnBegin= "function() {this.style.display='none';}"
                };

            // Render More Entries
            routeData["page"] = pager.PageIndex + 1;
            return helper.ActionLink("More Entries", "Index", routeData, options);
        }




    }
}
